YapDatabase/YapDatabase.m

```diff
@@ -151,8 +151,14 @@ + (YapDatabaseSerializer)defaultSerializer
 **/
 + (YapDatabaseDeserializer)defaultDeserializer
 {
-	return ^ id (NSString __unused *collection, NSString __unused *key, NSData *data){
-		return data && data.length > 0 ? [NSKeyedUnarchiver unarchiveObjectWithData:data] : nil;
+	return ^ id (NSString *collection, NSString *key, NSData *data){
+        NSError *error = nil;
+        id object = data && data.length > 0 ? [NSKeyedUnarchiver unarchiveTopLevelObjectWithData:data error:&error] : nil;
+        if (error)
```

Comment:
This should be checking `object` before it checks `error`.

```objc
if (object == nil && error != nil) {
    YDBLogError(@"Could not deserialize object in collection: %@ for key: %@ because of: %@", collection, key, error);
}
return object;
```

Reply:
If there's an error, regardless of whether there's an object, I think I'd like to see it. Do you disagree? I could change the log a bit to something like: `"Encountered error while deserializing..."`. If you feel strongly though I can check the object first.

Either way, given that this API seems to have been written for swift, I feel pretty good that object and error will never both be non-nil.

Reply:
Yeah for some reason I thought it was returning nil in the if block. S'all good.